src/io/overlapped.cr

```diff
@@ -77,7 +116,17 @@ module IO::Overlapped
 
   # Returns false the the operation timed out
   def schedule_overlapped(timeout : Time::Span?, line = __LINE__) : Bool
-    Crystal::EventLoop.wait_completion(timeout.try(&.total_milliseconds) || LibC::INFINITE)
+    if timeout
+      timeout_event = Crystal::Event.new(Fiber.current)
+      timeout_event.add(timeout)
+    else
+      timeout_event = Crystal::Event.new(Fiber.current, Time::Span::MAX)
+    end
+    Crystal::EventLoop.enqueue(timeout_event)
+
+    Crystal::Scheduler.reschedule
+
+    Crystal::EventLoop.dequeue(timeout_event)
```

Comment:
Won't `run_once` already dequeue? If not, seems weird that it dequeues sometimes, but not always.

Reply:
`run_once` deques when a timeout happens. This operations handles the case when there's no timeout, i.e. the operation completes in time and the fiber resumes. Then we need to dequeue the timeout event.